### DISCUSSION TITLE
Osu!lazer still open in background even after quitting

### DISCUSSION THREAD

**Discussion by Frous-s** [ppy/osu#15955] [Q&A]
osu lazer is still open in background after quitting, using up to 12% CPU until you force exit the task on task manager.

PC specs:
Ryzen 5 2600
16 GB DDR4 3200 MHz
RX 580 8 GB
 Windows 10 21H2
 just in case

**Comment by bdach** [comment]
Please attach logs from either:
* `%AppData%/osu/logs` (on Windows),
* `~/.local/share/osu/logs` (on Linux and macOS),
* `Android/Data/sh.ppy.osulazer/logs` (on Android).

**Comment by Frous-s** [comment]
[runtime.log](https://github.com/ppy/osu/files/7663309/runtime.log
[session.log](https://github.com/ppy/osu/files/7663314/session.log)
[update_success.log](https://github.com/ppy/osu/files/7663316/update_success.log)
)

**Reply by bdach** [reply]
these are logs from stable. please try again, and double-check the location.

**Reply by Frous-s** [reply]
the location of lazer doesn't look at all like the path you provided on widows
![image](https://user-images.githubusercontent.com/66706431/144929773-2d7c5710-624a-4cfa-b95a-1d129c36270f.png)

**Reply by Theighlin** [reply]
that is not the path he provided, that is the application folder. He was talking about this folder
![image](https://user-images.githubusercontent.com/53081750/144930373-06dda62a-5d7a-4cac-ae5c-f8be2096f76a.png)

**Comment by Frous-s** [comment]
[performance.log](https://github.com/ppy/osu/files/7677857/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7677858/runtime.log)
[updater.log](https://github.com/ppy/osu/files/7677859/updater.log)
[database.log](https://github.com/ppy/osu/files/7677860/database.log)
[network.log](https://github.com/ppy/osu/files/7677861/network.log)

Hopefully these are the files you're looking for.

**Reply by bdach** [reply]
these logs look clean as far as i can tell, doesn't look like anything out of the ordinary. we may need a process dump to proceed further. _**DO NOT**_ post one publically here, however, as it can reveal sensitive information.

**Reply by smoogipoo** [reply]
You can collect the dump via task manager and email it to pe@ppy.sh with a link to this discussion.

**Reply by peppy** [reply]
@Frous-s Any chance of getting a process dump? Are you seeing this happen every time you close the game, or just that once?

**Reply by Frous-s** [reply]
Hi, I reopened the game today. it's every time I close the game, currently there's two processes in background, one from updating the other one after some testing. I made a process dump, but the file is currently 2.8 GBs, so I'm letting you know beforehand since the file size is huge, this file size could give a clue regarding the problem, if the file is 100% necessary, I'm going to send it ASAP.

**Reply by peppy** [reply]
This really sounds like something on your PC is hooking the app and causing it to stay open.

**Reply by Frous-s** [reply]
could it be hawku drivers? there's really nothing that could really mess with lazer other than medalTV and maybe hawku driver.

**Reply by peppy** [reply]
more likely medalTV than a tablet "driver".

**Comment by NekkoDroid** [comment]
I've been also having this problem. lazer randomly (not every time) decides to stay open in the background when exiting.

I've tried to reproduce it while building, running and debugging lazer but I haven't been able to reproduce the problem when there was a debugger attached (did sometimes happen when running without the debugger).

**Reply by peppy** [reply]
If you can reproduce please follow the instructions to make a process dump https://github.com/ppy/osu/discussions/15955#discussioncomment-1774741

![Parallels Desktop 2022-03-17 at 06 44 46](https://user-images.githubusercontent.com/191335/158752156-26058324-1946-4364-bd79-daf5dc8b159b.png)
